xapian-bindings/java/SmokeTest.java

```diff
@@ -121,8 +121,19 @@ public static void main(String[] args) throws Exception {
 		System.exit(1);
 	    }
 	    Enquire enq = new Enquire(db);
-	    enq.setQuery(new Query(Query.OP_OR, "there", "is"));
+	    enq.setQuery(new Query("dog"));
 	    MSet mset = enq.getMSet(0, 10);
+	    // MSet size is 0
+	    if(mset.size() != 0) {
+	    	for(MSetIterator.MatchDescriptor descriptor : mset) {
+	    		System.err.println("Empty iterator entered invalid section (from MSet java iterator)");
```

Comment:
Smoketest should exit non-zero when it fails; because it's very simple that means a failure won't bother running the other tests. However that's better than having to read the output (for instance, it allows our buildbot to notice when tests fail, which helps keep track of problems).

jUnit will make this all more flexible and powerful, true :)